generator/ServiceClientGeneratorLib/Generators/Marshallers/JsonRPCResponseUnmarshaller.tt

```diff
@@ -157,20 +157,26 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
             if  (this.Config.ServiceModel.IsAwsQueryCompatible)
             {
 #>
-<# // Create a copy of context with headers in the response 
-
-#>
+<# // Create a copy of context with headers in the response #>
             using (var contextCopy = new JsonUnmarshallerContext(streamCopy, true, context.ResponseData))
+            {
+                StreamingUtf8JsonReader readerCopy = new StreamingUtf8JsonReader(streamCopy);
 <#
+                GenerateAWSQueryCompatibleBlock();
```

Comment:
Same comment here, wanted to check if removing `this.Config.ServiceModel.IsAwsQueryCompatible` is intentional.